ruma-events/src/key/verification/accept.rs

```diff
@@ -20,33 +22,235 @@ pub type AcceptEvent = BasicEvent<AcceptEventContent>;
+                key_agreement_protocol: KeyAgreementProtocol::Curve25519,
+                message_authentication_code: MessageAuthenticationCode::HkdfHmacSha256,
+                short_authentication_string: vec![ShortAuthenticationString::Decimal],
+                commitment: "test_commitment".into(),
+            }),
+        };
+
+        let key_verification_accept = AcceptEvent { content: key_verification_accept_content };
+
+        let json_data = json!({
+            "content": {
+                "transaction_id": "456",
+                "method": "m.sas.v1",
+                "commitment": "test_commitment",
+                "key_agreement_protocol": "curve25519",
+                "hash": "sha256",
+                "message_authentication_code": "hkdf-hmac-sha256",
+                "short_authentication_string": ["decimal"]
+            },
+            "type": "m.key.verification.accept"
+        });
+
+        assert_eq!(to_json_value(&key_verification_accept).unwrap(), json_data);
+
+        let json_data = json!({
+            "content": {
+                "transaction_id": "456",
+                "method": "m.sas.custom",
+                "test": "field",
+            },
+            "type": "m.key.verification.accept"
+        });
+
+        let key_verification_accept_content = AcceptEventContent {
+            transaction_id: "456".into(),
+            method: AcceptMethod::Custom(CustomContent {
+                method: "m.sas.custom".to_owned(),
+                fields: vec![("test".to_string(), Value::from("field"))]
+                    .into_iter()
+                    .collect::<BTreeMap<String, Value>>(),
+            }),
+        };
+
+        let key_verification_accept = AcceptEvent { content: key_verification_accept_content };
+
+        assert_eq!(to_json_value(&key_verification_accept).unwrap(), json_data);
+    }
+
+    #[test]
+    fn deserialization() {
+        let json = json!({
+            "transaction_id": "456",
+            "commitment": "test_commitment",
+            "method": "m.sas.v1",
+            "hash": "sha256",
+            "key_agreement_protocol": "curve25519",
+            "message_authentication_code": "hkdf-hmac-sha256",
+            "short_authentication_string": ["decimal"]
+        });
+
+        // Deserialize the content struct separately to verify `TryFromRaw` is implemented for it.
+        assert_matches!(
+            from_json_value::<Raw<AcceptEventContent>>(json)
+                .unwrap()
+                .deserialize()
+                .unwrap(),
+            AcceptEventContent {
+                transaction_id,
+                method: AcceptMethod::MSasV1(MSasV1Content {
+                        commitment,
+                        hash,
+                        key_agreement_protocol,
+                        message_authentication_code,
+                        short_authentication_string,
+                    }
+                )
```

Comment:
This is not indented correctly. Unfortunately, `rustfmt` doesn't know how to format `assert_matches!` calls so it mostly leaves them alone.